tests/it/util.rs

```diff
@@ -0,0 +1,119 @@
+use std::{
+    ffi::{OsStr, OsString},
+    path::{Path, PathBuf},
+};
+
+use assert_cmd::{assert::Assert, cargo::CargoError, Command};
+use assert_fs::{prelude::PathChild, TempDir};
+
+pub fn wasm_file(name: &str) -> PathBuf {
+    let path = Path::new(name);
+    let wasm = if path
+        .extension()
+        .map_or(false, |ext| ext.eq_ignore_ascii_case("wasm"))
+    {
+        ""
+    } else {
+        ".wasm"
+    };
+    PathBuf::from("target/wasm32-unknown-unknown/test-wasms").join(format!("{name}{wasm}"))
+}
+
+pub struct Soroban {
+    inner: Command,
+}
```

Comment:
❗ I think this type makes tradeoffs that compromise too much on the 1-1 mapping we currently have between the commands people will use, and the commands visible in the tests. The type introduces indirection. Without the type reading a test looks like a command that's being tested, so writing tests is in every also dogfooding. It's easy to look at the test and see that yes this is the exact command being executed. With the type the command being executed is being obfuscated and we invent a language of sorts the test writer is then interacting with rather than the raw command.

Before:
```rust
     const WASM: &str = "target/wasm32-unknown-unknown/test-wasms/test_invoker_account_exists.wasm";
     assert!(
         Path::new(WASM).is_file(),
         "file {WASM:?} missing, run 'make test-wasms' to generate .wasm files before running this test"
     );

     let tmp = TempDir::new().unwrap();
     let ledger = tmp.child("ledger.json");

     let mut cmd = Command::cargo_bin("soroban")?;
     cmd.arg("invoke");
     cmd.arg("--ledger-file").arg(ledger.as_os_str());
     cmd.arg("--id").arg("1");
     cmd.arg("--wasm").arg(WASM);
     cmd.arg("--fn").arg("invkexists");
     cmd.assert().success().stdout("true\n");
```

After:
```rust
     Soroban::invoke()?
         .ledger_file(temp_ledger_file())
         .contract_id("1")
         .wasm(wasm_file("test_invoker_account_exists"))?
         ._fn("invkexists")
         .assert()
         .success()
         .stdout("true\n");
```

Reply:
I just reorganized it so that there is a trait for creating the commands.  Thus turning into:

```rust
 Sandbox::new()
        .arg("invoke")
        .arg("--ledger-file")
        .arg(temp_ledger_file())
        .arg("--id=1")
        .arg("--wasm")
        .arg(test_wasm("test_invoker_account_exists"))
        .arg("--fn=invkexists")
        .assert()
        .success()
        .stdout("true\n");
```

Reply:
Is there a reason for breaking up the cmd from `.assert...`?